### ISSUE TITLE
Increase Header Size

### ISSUE THREAD

**Issue by juliofalbo** [tinyproxy/tinyproxy#446]
## Tinyproxy version
git master

## Issue
I'm having problems with a big cookie size, and as far I could see, Tinyproxy only handles headers until 1024 bytes (maybe I'm wrong).

So, my question is: Is there any configuration where I can increase this max amount?

Thanks.

**Comment by rofl0r** [comment]
> #define MAXIMUM_BUFFER_LENGTH (128 * 1024)

does changing (i.e. increasing) this value and recompiling tinyproxy change anything ?

**Comment by juliofalbo** [comment]
Yes, I did the test, and when I put `#define MAXIMUM_BUFFER_LENGTH (128 * 10240)`, I could do my request.

**Comment by rofl0r** [comment]
that's good to know, however i fear that we can't help here. having a single header line exceeding 128KB is beyond unreasonable. there are various limits in POSIX that define e.g. the maximum length of a single line in any text file to 4 KB (LINE_MAX). supporting anything greater than that might lead to security problems such as DOS attacks.
in this particular case, i would suggest contacting the upstream website's webmaster and inform him that these huge cookies break 3rd party applications, like tinyproxy.
if there's really a need to pack all this data into cookies, it might make sense to pack it with some sort of compression algorithm. but it would be better to just reduce the amount of data stored inside that cookie.
